### PR TITLE
feat(dapi)!: split getStatus into two endpoints

### PR DESCRIPTION
**Pull request by markin-io**
## Issue being fixed or feature implemented

Reworked getStatus query to not do Masternode related RPC reuqests 

## What was done?

Removed unnecessary and performance heavy `getMnSync('status')` and `getMasternode('status')` Core RPC calls from `getStatus` handler

## How Has This Been Tested?

CI

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [ ] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Comment by riongull** [comment]
Hey @markin-io could you give some more context about what this is about?  It's a pretty big PR.  What's the motivation, what are the removals being replaced with, etc.  Thanks!

**Review by shumkov — CHANGES_REQUESTED** [review]
I was speaking with some people and community and it seems masternode status is something what people is using. Shall we just move it to a separate endpoint instead of removing? `getMasternodeStatus`? In this case it's probably make sense to rename `getStatus` to `getBlockchainStatus`

File: packages/dapi-grpc/clients/core/v0/nodejs/CorePromiseClient.js [hunk 1]
```diff
@@ -24,8 +24,10 @@ const {
       platform: {
         dapi: {
           v0: {
-            GetStatusRequest: PBJSGetStatusRequest,
-            GetStatusResponse: PBJSGetStatusResponse,
+            GetCoreChainStatusRequest: PBJSGetCoreChainStatusRequest,
```
Comment (shumkov) [inline packages/dapi-grpc/clients/core/v0/nodejs/CorePromiseClient.js:27]: We already in Core service. There is no need to point to Core again. I think we agreed on `getBlockchainStatus` (similar to Core RPC command name).

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/dapi/lib/grpcServer/handlers/core/getBlockchainStatusHandlerFactory.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    time.setMedian(blockchainInfoResponse.mediantime);
+
+    const chain = new GetBlockchainStatusResponse.Chain();
+    chain.setName(blockchainInfoResponse.chain);
+    chain.setBlocksCount(blockchainInfoResponse.blocks);
+    chain.setHeadersCount(blockchainInfoResponse.headers);
+    chain.setBestBlockHash(Buffer.from(blockchainInfoResponse.bestblockhash, 'hex'));
+    chain.setDifficulty(blockchainInfoResponse.difficulty);
+    chain.setChainWork(Buffer.from(blockchainInfoResponse.chainwork, 'hex'));
+    chain.setIsSynced(blockchainInfoResponse.verificationprogress === 1);
```
Comment (shumkov) [inline packages/dapi/lib/grpcServer/handlers/core/getBlockchainStatusHandlerFactory.js:44]: Isn't it's a float?

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/dapi/lib/grpcServer/handlers/core/getBlockchainStatusHandlerFactory.js (hunk 2 above)
Comment (shumkov) [reply packages/dapi/lib/grpcServer/handlers/core/getBlockchainStatusHandlerFactory.js:44]: Oh, that's JS :))

**Review by shumkov — DISMISSED** [review]
👍

**Review by shumkov — APPROVED** [review]

**Pull request merged**